Sources/BAPromise/Promise.swift

```diff
@@ -560,3 +562,19 @@ extension Array where Element == Completable {
         return returnedPromise
     }
 }
+
+typealias dispatch_block_t_swift = () -> Void
+
+extension Thread {
+  
+  func runBlock(_ block: @escaping dispatch_block_t_swift) {
+      block()
+  }
+
+  func async(_ block: @escaping dispatch_block_t_swift) {
+    perform(Selector(("runBlock:")),
```

Comment:
At one point I had this in Swift. I was getting intermittent memory failures, though, and moved it back to ObjC. I'll dig up the commit and see if I can remember why.

That said, Realm 5 _finally_ introduced Queue-specific realm objects in addition to the Thread-specific realm objects, so my primary need for all this thread nonsense could go away soon.

Reply:
This was the commit where I had stopped using Swift for this. I think that the Swift compiler was doing the wrong thing with `block` and leading to occasional memory failures (use after release)
https://github.com/benski/BAPromise/commit/a13148e73d701659c04160c21eb08f62efc8746b

Reply:
Might be fixed now with swift 5.3? Release notes mention runtime memory improvements. 
Fixed the project file!!